src/components/BookmarksBar/index.tsx

```diff
@@ -24,25 +23,21 @@ const BookmarksBar: React.FC<BookmarksBarProps> = ({
   scrollToLine,
 }) => {
   const { sendEvent } = useLogWindowAnalytics();
-
-  const [selectedLine] = useQueryParam<number | undefined>(
-    QueryParams.SelectedLine,
-    undefined
-  );
-  const [bookmarks, setBookmarks] = useQueryParam<number[]>(
-    QueryParams.Bookmarks,
-    []
-  );
+  const { selectedLine, bookmarks, setBookmarks } = useLogContext();
 
   // Set the initial bookmarks on load.
   useEffect(() => {
-    if (bookmarks.length === 0 && lineCount !== 0) {
-      if (lineCount === 1) {
-        setBookmarks([0]);
-      } else {
-        setBookmarks([0, lineCount - 1]);
+    // This is a workaround for a known issue in Jest tests.
+    // https://github.com/remix-run/react-router/issues/9314
+    setTimeout(() => {
+      if (bookmarks.length === 0 && lineCount !== 0) {
+        if (lineCount === 1) {
+          setBookmarks([0]);
+        } else {
+          setBookmarks([0, lineCount - 1]);
+        }
       }
-    }
+    }, 0);
```

Comment:
One of the things I did in this refactor was centralize the URL params in `LogContext`, which causes this issue. :/ 

I wonder if I shouldn't put them in the `LogContext`, but having them in one place can also be useful... Another option is to move this URL update into `LogContext`, but then it's not testable as something that happens in `BookmarksBar`.

Reply:
I'm not to big of a fan of this approach if the only reason is to appease jest.